src/transformers/models/relation_detr/modeling_relation_detr.py

```diff
@@ -0,0 +1,2788 @@
+
+            model._modules[name] = new_module
+
+        if len(list(module.children())) > 0:
+            replace_batch_norm(module)
+
+
+class RelationDetrConvEncoderPostProcess(nn.Module):
+    """Post process for the ConvEncoder. This is used to be compatible with the FocalNet backbone in official repo."""
+
+    def __init__(self, in_channels: int, post_layer_norm: bool = False):
+        super().__init__()
+        self.in_channels = in_channels
+        self.post_layer_norm = post_layer_norm
+        if self.post_layer_norm:
+            for idx, channel in enumerate(in_channels):
+                self.add_module(f"norm{idx}", nn.LayerNorm(channel))
+
+    def forward(self, multi_level_feats: List[Tensor]):
+        if self.post_layer_norm:
+            # convert N C H W to N H W C
+            if all(feat.shape[1] == channel for feat, channel in zip(multi_level_feats, self.in_channels)):
+                multi_level_feats = [feat.permute(0, 2, 3, 1) for feat in multi_level_feats]
+
+            for idx, feat in enumerate(multi_level_feats):
+                feat = getattr(self, f"norm{idx}")(feat)
+                multi_level_feats[idx] = feat
+
+        # convert N H W C to N C H W
+        if all(feat.shape[-1] == channel for feat, channel in zip(multi_level_feats, self.in_channels)):
+            multi_level_feats = [feat.permute(0, 3, 1, 2) for feat in multi_level_feats]
+
+        return multi_level_feats
+
+
+class RelationDetrConvEncoder(nn.Module):
+    """
+    Convolutional backbone, using either the AutoBackbone API or one from the timm library.
+
+    nn.BatchNorm2d layers are replaced by RelationDetrFrozenBatchNorm2d as defined above.
+
+    """
+
+    def __init__(self, config):
+        super().__init__()
+
+        self.config = config
+
+        # For backwards compatibility we have to use the timm library directly instead of the AutoBackbone API
+        if config.use_timm_backbone:
+            # We default to values which were previously hard-coded. This enables configurability from the config
+            # using backbone arguments, while keeping the default behavior the same.
+            requires_backends(self, ["timm"])
+            kwargs = getattr(config, "backbone_kwargs", {})
+            kwargs = {} if kwargs is None else kwargs.copy()
+            out_indices = kwargs.pop("out_indices", (2, 3, 4) if config.num_feature_levels > 1 else (4,))
+            num_channels = kwargs.pop("in_chans", config.num_channels)
+            if config.dilation:
+                kwargs["output_stride"] = kwargs.get("output_stride", 16)
+            backbone = create_model(
+                config.backbone,
+                pretrained=config.use_pretrained_backbone,
+                features_only=True,
+                out_indices=out_indices,
+                in_chans=num_channels,
+                **kwargs,
+            )
+        else:
+            backbone = load_backbone(config)
```

Comment:
We don't need everything that is provided for backward compat, please cut these lines of code. It should be just `backbone = load_backbone(config)`